index.js

```diff
@@ -0,0 +1,29 @@
+const { parse } = require('csv-parse')
+const fs = require('fs')
+const habitablePlanets = []
+
+function isHabitable (planet) { // checks if all the conditions to confirm that a planet is habitable are met
+  return planet.koi_disposition === 'CONFIRMED' &&
+    planet.koi_insol > 0.36 && planet.koi_insol < 1.11 &&
+    planet.koi_prad > 1.6
+}
+
+fs.createReadStream( // transforms the kepler_data.csv file into a readable stream
+  'kepler_data.csv'
+).pipe(parse({ // parses the stream into JSON
+  comment: '#',
+  columns: true
+})
+).on('data', (data) => {
+  if (isHabitable(data)) {
+    habitablePlanets.push(data) // filters the data based on the habitability criteria and pushes each chunk into the habitablePlanets array

```

Comment:
i dont thinks this filters the array it just pushes. I think we can remove all the comments.

This is a good article : https://medium.com/codex/clean-code-comments-833e11a706dc

Reply:
It doesn't filter the array, It filters each chunk of data to find habitable planets, then it pushes into the habitablePlanets array.